File: AOJ/0316/fishing_competition.java
import java.util.Scanner;

class Main {
  int getScore(int iwa, int yam,
		 int a, int b, int c, int d) {
    return iwa * a + yam * b +
	(iwa / 10) * c + (yam / 20) * d;  
  }
  
  void solve() {
    Scanner sc = new Scanner(System.in);
    int h1 = sc.nextInt(), h2 = sc.nextInt();
    int k1 = sc.nextInt(), k2 = sc.nextInt();
    int a = sc.nextInt(),  b = sc.nextInt();
    int c = sc.nextInt(),  d = sc.nextInt();

    int hiro = getScore(h1, h2);
    int ken = getScore(k1, k2);
	
    if      ( hiro > ken )
      System.out.println("hiroshi");
    else if ( hiro < ken )
      System.out.println("kenjiro");
    else
      System.out.println("even");
  }

  public static void main(String[] args) {
      new Main().solve();
  }
}

File: AOJ/0316/fishing_competition.py
def getScore(iwa, yam, a, b, c, d):
    return iwa * a + yam * b + \
        (iwa // 10) * c + (yam // 20) * d;

h1, h2 = map(int, input().split())
k1, k2 = map(int, input().split())
a, b, c, d = map(int, input().split())
hiro = getScore(h1, h2, a, b, c, d)
ken = getScore(k1, k2, a, b, c, d)

if   hiro > ken: print("hiroshi")
elif hiro < ken: print("kenjiro")
else: print("even")
